pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -568,25 +572,25 @@ func (r *PerformanceProfileReconciler) applyComponents(profile *performancev2.Pe
 	}
 
 	// get mutated machine config
-	mcMutated, err := r.getMutatedMachineConfig(context.TODO(), components.MachineConfig)
+	mcMutated, err := resources.GetMutatedMachineConfig(context.TODO(), r.Client, components.MachineConfig)
```

Comment:
let's create a single context - TODO() is ok and let's reuse alongside the client for all the calls in the function

Reply:
I'm planning on doing it on a later PR. (part 2 of hypershift adoption)

Reply:
ok, fair enough